acados/ocp_nlp/ocp_nlp_dynamics_cont.c

```diff
@@ -527,24 +527,13 @@ void ocp_nlp_dynamics_cont_update_qp_matrices(void *config_, void *dims_, void *
 
     if (opts->compute_hess)
     {
-        int hess_index = 0;
-        for (int j = 0; j < nx; j++) {
-            for (int k = j; k < nx; k++) {
-                BLASFEO_DMATEL(&mem->hes, nu+k, nu+j) = work->sim_out->S_hess[hess_index];
-                hess_index++;
-            }
-            for (int k = 0; k < nu; k++) {
-                // S: nu x nx
-                BLASFEO_DMATEL(&mem->hes, nu+j, k) = work->sim_out->S_hess[hess_index];
-                hess_index++;
-            }
-        }
-        for (int j = 0; j < nu; j++) {
-            for (int k = j; k < nu; k++) {
-                BLASFEO_DMATEL(&mem->hes, k, j) = work->sim_out->S_hess[hess_index];
-                hess_index++;
-            }
-        }
+        // unpack d*_d2u
+        blasfeo_pack_dmat(nu, nu, &work->sim_out->S_hess[(nx+nu)*nx + nx], nx + nu,
+                                     &mem->hes, 0, 0);
+        // unpack d*_dux: mem-hess: nx x nu
+        blasfeo_pack_dmat(nx, nu, &work->sim_out->S_hess[(nx + nu)*nx], nx + nu, &mem->hes, nu, 0);
+        // unpack d*_d2x
+        blasfeo_pack_dmat(nx, nx, &work->sim_out->S_hess[0], nx + nu, &mem->hes, nu, nu);
```

Comment:
Thanks for simplifying this! I did not check the details, but do you have a (general enough) test problem to confirm that the two pieces of code do the same thing? Also what about @giaf's comment on directly adding this to RSQrq instead? Maybe clarify with @roversch before we forget)

Reply:
I tested with the `pendulum_sqp` example, in cooperation with @roversch 
@dkouzoup seems I missed that comment...
anyway the interface to hessians is cleaner now, and it can be used in the higher level moduls.

Reply:
The two pieces of code do not look like doing the same thing: in the first it looks like only the data of the lower-triangular matrices are stored (contiguously) in S_hess, while in the second it looks like the entire matrices are stored in S_hess (also with the unreferenced upper triangular part). Which one is the correct one? How is data stored in S_hess?